synology_dsm/api/surveillance_station/__init__.py

```diff
@@ -0,0 +1,122 @@
+"""Synology SurveillanceStation API wrapper."""
+import urllib
+
+from .camera import SynoCamera
+from .const import MOTION_DETECTION_BY_SURVEILLANCE, MOTION_DETECTION_DISABLED
+
+
+class SynoSurveillanceStation(object):
+    """An implementation of a Synology SurveillanceStation."""
+
+    API_KEY = "SYNO.SurveillanceStation.*"
+    CAMERA_API_KEY = "SYNO.SurveillanceStation.Camera"
+    CAMERA_EVENT_API_KEY = "SYNO.SurveillanceStation.Camera.Event"
+    HOME_MODE_API_KEY = "SYNO.SurveillanceStation.HomeMode"
+    SNAPSHOT_API_KEY = "SYNO.SurveillanceStation.SnapShot"
+
+    def __init__(self, dsm):
+        """Initialize a Surveillance Station."""
+        self._dsm = dsm
+        self._cameras_by_id = {}
+
+    def update(self):
+        """Update cameras and motion settings with latest from API."""
+        response = self._dsm.get(self.CAMERA_API_KEY, "list")
+        for data in response["data"]["cameras"]:
+            if data["id"] in self._cameras_by_id:
+                self._cameras_by_id[data["id"]].update(data)
+            else:
+                self._cameras_by_id[data["id"]] = SynoCamera(data)
+
+        for camera_id in self._cameras_by_id:
+            self._cameras_by_id[camera_id].update_motion_detection(
+                self._dsm.get(
+                    self.CAMERA_EVENT_API_KEY, "MotionEnum", {"camId": camera_id}
+                )["data"]
+            )
+
+        live_view_response = self._dsm.get(
+            self.CAMERA_API_KEY,
+            "getLiveViewPath",
+            {"idList", self._cameras_by_id.keys()},
+        )
+        for live_view_data in live_view_response["data"]:
+            self._cameras_by_id[live_view_data["id"]].live_view.update(live_view_data)
+
+    # Camera
+    def get_all_cameras(self):
+        """Return a list of cameras."""
+        return self._cameras_by_id.values()
+
+    def get_camera(self, camera_id):
+        """Return camera matching camera_id."""
+        return self._cameras_by_id[camera_id]
+
+    def get_camera_live_view_path(self, camera_id, video_format=None):
+        """Return camera live view path matching camera_id (video_format: mjpeg_http | multicast | mxpeg_http | rtsp_http | rtsp)."""
+        if video_format:
+            return getattr(self._cameras_by_id[camera_id].live_view, video_format)
+        return self._cameras_by_id[camera_id].live_view
+
+    def get_camera_image(self, camera_id):
+        """Return bytes of camera image for camera matching camera_id."""
+        return self._dsm.get(
+            self.CAMERA_API_KEY, "getSnapshot", {"cameraId": camera_id}
```

Comment:
```suggestion
            self.CAMERA_API_KEY, "GetSnapshot", {"id": camera_id}
```

Reply:
Nice you see that !

Because I checked the doc : https://github.com/ProtoThis/python-synology#official-references --> Surveillance Station

In "2.3.4.5 GetSnapshot method" paragraph, it is saying `id`.

But once, there is an exemple with `cameraId`: `"snapshot_path": "/webapi/_______________________________________________________entry.cgi?
api=SYNO.SurveillanceStation.Camera&method=GetSnapshot&version=1&cameraId=70×tamp=142491912
9&preview=true"` and also the base library: 

https://github.com/snjoetw/py-synology/blob/eaf7aab9b5963bd529b3f805ffd9db7ad19b3e6d/synology/api.py#L179-L191

Reply:
@Quentame Tested again and this seems to be a new change in API 9. Previously, it uses `cameraId`.

Reply:
So I'll add both for beeing old version compat.

or maybe use the value of `snapshot_path` provided with Camera `GetInfo`